modules/finality-verifier/src/lib.rs

```diff
@@ -141,7 +141,7 @@ pub mod pallet {
 				<Error<T>>::InvalidAncestryProof
 			);
 
-			T::HeaderChain::append_header(finality_target);
+			let _ = T::HeaderChain::append_header(finality_target)?;
```

Comment:
I've probably missed some substrate updates, but last time I was checking, returning error from extrinsic **after** it  has made some storage changes wasn't considered a good practice (because storage changes aren't reverted in this case). So `RequestCount` will be increased despite execution failure. This may allow malicious relayer to submit invalid headers over and over again, blocking pallet from progressing.

Reply:
Yeah, this is a problem whose scope is out side of this PR: https://github.com/paritytech/srlabs_findings/issues/47#issuecomment-779096609.

Reply:
> Actaully Slava has a good point here, it seems it was missed in previous reviews but a failed extrinsic (returning an error) MUST NOT alter the storage. It's recommended to either use transactional storage here or move storage modifications after all verifications are done.

I was aware that it was writing to storage despite having failed (specifically having failed to verify justifications, since that used to be the only exit point to the extrinsic). What I had in mind was that _any request_ that was made should count towards the request count. 

> Regarding the audit issue itself, since we don't import ancestry it doesn't matter how many blocks get finalized (indirectly). We should be limitting how fast the on-chain storage grows.

You make a good point here. The real goal here isn't to limit the number of requests, but instead to limit the _storage_ growth. In our case storage doesn't grow if somebody sends invalid data, so it doesn't matter if we don't keep track of that request. 

So I think it was my fault since I lost sight of the true goal of the rate limiter.

I will address this in a different PR since it's distinct from the ancestry checker changes in this PR.

Reply:
I've opened #765.